src/Orbits.jl

```diff
@@ -1,194 +1,385 @@
+    if m_1f == -1
+        m_1f = m_1i
+    end
+    mtilde = (m_1f+m_2f)/(m_1i+m_2i) 
+    a_i = kepler_a_from_P(;m_1=m_1i, m_2=m_2i, P=P_i)
+    v_rel = average_orbital_velocity(;m_1=m_1i, m_2=m_2i, a=a_i)
+    vkick_div_vrel = vkick/v_rel
+    vimp_div_vrel = vimp/v_rel
+    # convert trig functions to vars
+    cosθ = cos(θ)
+    sinθ = sin(θ)
+    cosϕ = cos(ϕ)
+    sinϕ = sin(ϕ)
+    ξ = (1 + vkick_div_vrel^2 + vimp_div_vrel^2 + 2*vkick_div_vrel*cosθ - 2*vimp_div_vrel*sinθ*cosϕ)/(mtilde)
+    if (ξ>2)
         return (NaN, NaN)
     end
-    Pfinal = P/sqrt(mtilde*(2-xi)^3)
-    efinal = sqrt(1+(xi-2)*(Q+1))
-    return (Pfinal, efinal)
-end
+    tanγ = (vkick*sinθ*cosϕ - vimp)/sqrt((v_rel + vkick*cosθ)^2 + vkick^2*sinθ^2*sinϕ^2)
+    cosγ = 1/sqrt(1+tanγ^2)
+
+    # Orbital parameters
+    P_f = P_i/sqrt(mtilde*(2-ξ)^3) # RTW check this
+    e_f = sqrt(1 + ξ*(ξ-2)*cosγ^2)
+
+    #Q = (ξ-1) - (vkick_div_vrel*sinθ*cosϕ - vimp_div_vrel)^2/(mtilde) 
+    #e_f_old = sqrt(1+(ξ-2)*(Q+1)) # RTW where does this come from?
+    # RTW TODO test both values for e_f
 
-function systemic_velocity(vrel,m1_i,m2_i,m2_f,vkick,θ,ϕ)
-    Δp_x = (m2_f*m1_i-m2_i*m1_i)/(m2_i+m1_i)*vrel+m2_f*vkick*cos(θ)
-    Δp_y = m2_f*vkick*sin(θ)*cos(ϕ)
-    Δp_z = m2_f*vkick*sin(θ)*sin(ϕ)
-    return sqrt(Δp_x^2+Δp_y^2+Δp_z^2)/(m2_f+m1_i)/1e5
+    return (P_f, e_f)
 end
 
+
+"""
+    post_supernova_circular_orbit_vsys(;m_1i, m_2i, a_i, m_1f=-1, m_2f, vkick=0, θ=0, ϕ=0, vimp=0)
+
+Compute post-kick properties for a circular pre-explosion orbit using equations from Tauris & Takens (1999)
+
+# Arguments:
+- m_1i:  pre-explosion  mass of non-exploding component [Msun]           
+- m_2i:  pre-explosion  mass of exploding component     [Msun]       
+- a_i:   pre-explosion orbital separation               [cm]
+- m_1f:  post-explosion mass of non-exploding component [Msun]           
+- m_2f:  post-explosion mass of exploding component     [Msun]   
+- vkick: kick velocity                                  [km/s] 
+- θ:     polar kick angle (away from e_par)             [rad]
+- ϕ:     azimuthal kick angle (off of e_perp)           [rad]
+- vimp:  imparted kick velocity on companion            [km/s]     
+    
+# Output:
+- vsys_f: post-explosion systemic velocity            [km/s]
 """
-    RV_semiamplitude_K(P, e, i, m1, m2)
+function post_supernova_circular_orbit_vsys(;m_1i, m_2i, a_i, m_1f=-1, m_2f, vkick=0, θ=0, ϕ=0, vimp=0)
+    if m_1f == -1
+        m_1f = m_1i
+    end
+    mtilde = (m_1f+m_2f)/(m_1i+m_2i) 
+    v_rel = average_orbital_velocity(;m_1=m_1i, m_2=m_2i, a=a_i)
+    vkick_div_vrel = vkick/v_rel
+    vimp_div_vrel = vimp/v_rel
+    # convert trig functions to vars
+    cosθ = cos(θ)
+    sinθ = sin(θ)
+    cosϕ = cos(ϕ)
+    sinϕ = sin(ϕ)
+    ξ = (1 + vkick_div_vrel^2 + vimp_div_vrel^2 + 2*vkick_div_vrel*cosθ - 2*vimp_div_vrel*sinθ*cosϕ)/(mtilde)
+    if (ξ>2)
+        return NaN 
+    end
```

Comment:
I don't think we need to do this check, this is will add up to computational cost and is repetition. One would expect the user to check if the binary is bound by running the other function first. In any case, the result will be a physical one anyhow. It tells you the velocity of the center of mass of the system, which is well defined even if the binary is unbound

Reply:
I am happy to get rid of the check. But most of this function (`post_supernova_circular_orbit_vsys`), is repeated in the functions above it (`post_supernova_circular_orbit_a` and `post_supernova_circular_orbit_P`). I had thought this was because we may only call one of these functions at a time, so you don't want to waste CPU on all the lines that are unique to each function. But if we're going to call more than one of these anyway in a typical use case (say, `post_supernova_circular_orbit_P` and `post_supernova_circular_orbit_vsys`), then we should consolidate them into one.

Reply:
I think for now we can just remove these lines. As I said the end result will still be a physical property. We can take care of further cleaning/consolidation later on